implement support for sub-menues (bsc #906990)

wfeldt opened the pull request:
See test/menu/isolinux.cfg for a sample config with two nested sub-menues.

wfeldt committed:
implement support for sub-menues (bsc #906990)

wfeldt commented:
There's a sample iso to try at ~snwint/Export/menu.iso.

jreidinger commented on themes/openSUSE/src/submenu.inc:
```diff
@@ -0,0 +1,122 @@
+/submenu.tag "->" def
+
+% submenues are stacked here
+/submenu.list 8 array def
```
does it mean, that number of submenu elements are limited to 8? if so, is it documented somewhere?

wfeldt replied:
The nesting level is 8; it will just refuse to enter deeper submenues. I think this is good enough...

jreidinger replied:
I am also fine with such limit, I just think we should document it somewhere else, not just burden it inside code.

jreidinger commented:
Note related to question about limitation. Where it is documented? ( beside comment above code? ) I think it make sense to have some documentation how cfg look like and what is its limitation.

wfeldt committed:
- documented submenu syntax
- show submenu indicator in RTL layout
- on leaving a submenu, restore cursor position

jreidinger commented:
thanks for documentation. LGTM
